---
title: "More 10T"
output:
  html_document:
    toc: yes
    toc_float: yes
    number_sections: false
---
```{r load data, include = F, warning = F}
library(plotly)
library(heatmaply)
library(ggplot2)
library(gplots)
library(viridis)
library(UpSetR)
```


The other page is getting long.

## PCA with top 500 most variable {.tabset .tabset-pills .tabset-fade}
Taking the top 500 most variable genes among the samples and generating a regular old Euclidean distance matrix.

### Annot+TE
```{r tripca1, echo = F, warning = F}
load("D:/PhD/Analysis/10T/reinnier.RData")
All_HM

```

### Annot only
```{r tripca2, echo = F, warning = F}
ENS_HM
```

### TE only
```{r tripca3, echo = F, warning = F}
TE_HM
```

## {.unlisted .unnumbered}

<details>
<summary>Size Factors</summary>
**Size factor all entries**
```
                   dds.sizeFactor
H33K36M_C3H10T1_1       1.9754626
H33K36M_C3H10T1_2       0.7887794
H33K36M_C3H10T1_3       0.3056923
H33K36R_C3H10T1_1       1.3801394
H33K36R_C3H10T1_2       0.3812469
H33K36R_C3H10T1_3       0.3979667
NSD12DKO_C3H10T1_1      1.9744437
NSD12DKO_C3H10T1_2      0.7853220
NSD12DKO_C3H10T1_3      0.6446567
PA_C3H10T1_1            2.1517784
PA_C3H10T1_2            0.3786161
PA_C3H10T1_3            0.8761740
SETD2KO_C3H10T1_1       2.3758762
SETD2KO_C3H10T1_2       0.6693181
SETD2KO_C3H10T1_3       0.4933132
TKO_C3H10T1_1           2.7458456
TKO_C3H10T1_2           2.6752406
TKO_C3H10T1_3           2.9033106
```

**Size factor annot only**
```
                   ddsSub.sizeFactor
H33K36M_C3H10T1_1          1.9958859
H33K36M_C3H10T1_2          0.7398523
H33K36M_C3H10T1_3          0.2721026
H33K36R_C3H10T1_1          1.3409255
H33K36R_C3H10T1_2          0.3396584
H33K36R_C3H10T1_3          0.3556429
NSD12DKO_C3H10T1_1         1.9671956
NSD12DKO_C3H10T1_2         0.7176141
NSD12DKO_C3H10T1_3         0.5812148
PA_C3H10T1_1               2.1461076
PA_C3H10T1_2               0.3208860
PA_C3H10T1_3               0.7840284
SETD2KO_C3H10T1_1          2.3940446
SETD2KO_C3H10T1_2          0.6030431
SETD2KO_C3H10T1_3          0.4408191
TKO_C3H10T1_1              4.0540994
TKO_C3H10T1_2              4.1465894
TKO_C3H10T1_3              4.1640333 
```

</details>

## Normalized using size factors of only protein coding genes {#DEG}
`grep 'protein_coding' annotation.gtf | perl -ne '/^.+?\tgene\t.+?"(ENSMUS.+?)".+$/ && print $1'`

```{r sf, echo = F, warning = F}
load("D:/PhD/Analysis/10T/C3hsizeFactor.RData")
buttonplotSF

```

## Intergenic TEs Only
Graph of all the differentially expressed TEs, either filtered by the > 2 log2FoldChange cutoff.  The RefSeq annotations (ENSMUS) are not included in these graphs so note the different y axis values compared to the [graph above](#DEG)

```{r intergenic, echo = F, warning = F}
#  or no filter. I think the no filter list of genes may also be of interest especially of the expression of those TEs were particularly high.
buttonplotIN
```

### Intersection
Wow there are acually 25 TEs that were differentially expressed at a level greater than 2 LFC between all the pairwise comparisons. What are those 25?

```{r setsplot, echo = F, warning = F}
upset(fromList(DEG_TEs), order.by = "degree")
```

<details>
<summary>H33K36M & H33K36R & SETD2KO & NSD12DKO & TKO</summary>
```{r inter, size = "small", echo = F}
knitr::kable(IN_full[,-1])
```
</details>
  
<details>
<summary>H33K36M & H33K36R & SETD2KO & NSD12DKO</summary>
```{r inter4, size = "small", echo = F}
knitr::kable(IN_4[,-1])
```

-----  


## Genic TEs Only
There are some interesting observations about those genic TEs. The TEs below were sorted by whether they were in the reverse reading frame of the gene that they overlap with. I.e "No" = Same reading frame as overlapping gene, "Yes" = Opposite reading frame as overlapping gene. Not Only that, since there are sometimes genes that run both ways while overlapping each other, if at least one of the genes is on the opposite reading frame as the TE, the TE would be classified as a "Yes". Despite this, there seems to be more down-regulated TEs that are on the same reading frame as a gene and more up-regulated TEs that are on the opposite reading frame of a gene.

```{r genic TEs, echo = F}
load("D:/PhD/Analysis/10T/data/genicLFC0.5.RData")
test %>% ggplot(aes(x = Reversed, y = log2FoldChange, 
                    color = interaction(ifelse(log2FoldChange>0, "Pos", "Neg"), Reversed))) + 
  geom_jitter(width = 0.3) +
  facet_wrap(~Sample, nrow = 1) +
  theme_bw() + 
  labs(y = "Log2 Fold Change") +
  scale_color_manual(labels = c("No:LFC < -0.5", "No:LFC > 0.5", "Yes:LFC < -0.5", "Yes:LFC > 0.5"),
                     values = c("#7bb3e0", "#1e7cc7", "#db8f51", "#c25917")) + 
  theme(legend.title = element_blank())
```

In H33K36R there is an outlier with a great increase in fold change (>20 log2FC), this TE is on the opposite direction of nearby genes *Suv39h2*, *Gm13184* and *Gapvd1*. *Suv39H2* seems to be a H3K9me3 methyltransferase that uses monomethyl H3K9 as a substrate. IDK what are the implications of this, would be interesting to look at the histone methylation patterns around this area (as well as the raw reas to see if it's because of lack of coverage).

I tried looking at the GO enrichment of the nearby genes in each quadrant and **Alternative splicing** seems to come up in almost every group. Given that is a common GO term and is probably associated with a lot of genes.  
